src/rpc/util.h

```diff
@@ -329,6 +329,11 @@ class RPCHelpMan
 public:
     RPCHelpMan(std::string name, std::string description, std::vector<RPCArg> args, RPCResults results, RPCExamples examples);
     using RPCMethod = std::function<UniValue(const RPCHelpMan&, const JSONRPCRequest&)>;
+    struct HiddenArg {
```

Comment:
In commit " rpc: Assert that passed arg names are equal to hardcoded ones (server)" (fac2edbd5fc8075647e90e90d49b3cd910bdbbb7)

It seems like it might be cleaner if instead of adding a new HiddenArg struct, and new RPCHelpMan constructor and special case code for m_hidden_arg in GetArgNames, this just added a new `RPCArg::Type::HIDDEN` type. This would also support multiple hidden args, or hidden args after non-hidden args.

Reply:
Hidden args should not be a thing, because I can't imagine a use case for them.

If they are needed for testing, a new (hidden) RPC method with un-hidden RPC arguments should be added.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/19386#discussion_r447851960

> Hidden args should not be a thing, because I can't imagine a use case for them.
> 
> If they are needed for testing, a new (hidden) RPC method with un-hidden RPC arguments should be added.

Beyond testing, undocumented hidden args also seem useful as a way of removing options we no longer want to support like wallet account and min_conf arguments in some places, and still maintaining compatibility when arguments are set to specific values (e.g. min_conf=1, account="*").

But regardless of whether hidden args will be used in the future, adding a `Type::HIDDEN` enum value seems like a simpler way to implement hidden args now than adding a `HiddenArg` struct and alternate code path.

Reply:
> Beyond testing, undocumented hidden args also seem useful as a way of removing options

I disagree. The RPC interface seems the wrong place to hide options in the documentation but then still accept them. For deprecation we use `-rpcdeprecated` and proper updates to the documentation to say an argument has been deprecated or is discouraged.

I plan to remove `HiddenArg` before the next release, so the implementation shouldn't matter as long as it is minimal and correct.